crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -0,0 +1,381 @@
+			.unwrap_or(DEFAULT_SPEC_NAME);
+		let raw_chain_spec =
+			generate_raw_chain_spec(&binary_path, &plain_chain_spec, &raw_spec_name)?;
+		generated_files.push(format!(
+			"Raw chain specification file generated at: {}",
+			raw_chain_spec.display()
+		));
+
+		// Generate genesis artifacts.
+		if self.genesis_code {
+			spinner.set_message("Generating genesis code...");
+			let wasm_file_name = format!("para-{}.wasm", para_id);
+			let wasm_file = export_wasm_file(&binary_path, &raw_chain_spec, &wasm_file_name)?;
+			generated_files.push(format!(
+				"WebAssembly runtime file exported at: {}",
+				wasm_file.display().to_string()
+			));
+		}
+
+		if self.genesis_state {
+			spinner.set_message("Generating genesis state...");
+			let genesis_file_name = format!("para-{}-genesis-state", para_id);
+			let genesis_state_file =
+				generate_genesis_state_file(&binary_path, &raw_chain_spec, &genesis_file_name)?;
+			generated_files.push(format!(
+				"Genesis State file exported at: {}",
+				genesis_state_file.display().to_string()
+			));
+		}
+
+		console::Term::stderr().clear_last_lines(1)?;
+		let generated_files: Vec<_> = generated_files
+			.iter()
+			.map(|s| style(format!("{} {s}", console::Emoji("●", ">"))).dim().to_string())
+			.collect();
+		cli.success(format!("Generated files:\n{}", generated_files.join("\n")))?;
+		cli.outro(format!(
+			"Need help? Learn more at {}\n",
+			style("https://learn.onpop.io").magenta().underlined()
+		))?;
+
+		Ok("spec")
+	}
+}
+
+/// Guide the user to generate their chain specification.
+async fn guide_user_to_generate_spec() -> anyhow::Result<BuildSpecCommand> {
+	Cli.intro("Generate your chain spec")?;
+
+	// Confirm output path
+	let output_file: String = input("Name of the chain spec file. If a path is given, the necessary directories will be created:")
+		.placeholder("./template-chain-spec.json")
+		.default_input("./template-chain-spec.json")
+		.interact()?;
+
+	// Prompt for chain id.
+	let para_id: u32 = input("What parachain ID should the build use?")
+		.placeholder("2000")
+		.default_input("2000")
+		.interact()?;
+
+	// Prompt for relay chain.
+	let mut prompt =
+		cliclack::select("Choose the relay chain your chain will be connecting to: ".to_string());
+	for (i, relay) in RelayChain::VARIANTS.iter().enumerate() {
+		if i == 0 {
+			prompt = prompt.initial_value(relay);
+		}
+		prompt = prompt.item(
+			relay,
+			relay.get_message().unwrap_or(relay.as_ref()),
+			relay.get_detailed_message().unwrap_or_default(),
+		);
+	}
+	let relay_chain = prompt.interact()?.clone();
+
+	// Prompt for chain type.
+	// If relay is Kusama or Polkadot, then Live type is used and user is not prompted.
+	let chain_type: ChainType;
+	if relay_chain == RelayChain::Polkadot || relay_chain == RelayChain::Kusama {
+		chain_type = ChainType::Live;
+	} else {
+		let mut prompt = cliclack::select("Choose the chain type: ".to_string());
+		for (i, chain_type) in ChainType::VARIANTS.iter().enumerate() {
```

Comment:
Should we maybe filter out the chain types based on relay selected? Justification is preventing user from making a mistake.